src/script/chem/molfile/molfile.js

```diff
@@ -286,9 +286,135 @@ Molfile.prototype.writeCTab3000 = function (molecule) { // eslint-disable-line m
 	this.writeCR(`M  V30 COUNTS ${noOfAtoms} ${noOfBonds} ${noOfSGroups} 0 ${isChiral}`);
 	this.writeAtomBlock3000(molecule);
 	this.writeBondBlock3000(molecule);
+	this.writeSGroupBlock3000(molecule);
 	this.writeCR('M  V30 END CTAB');
 };
 
+Molfile.prototype.writeSGroupBlock3000 = function (molecule) {
+	if (molecule.sgroups.size === 0) {
+		return;
+	}
+	this.writeCR('M  V30 BEGIN SGROUP');
+	var sgmap = {};
+	var cnt = 1;
+	var sgmapback = {};
+	var sgorder = molecule.sGroupForest.getSGroupsBFS();
+	sgorder.forEach((id) => {
+		sgmapback[cnt] = id;
+		sgmap[id] = cnt++;
+	});
+	for (var q = 1; q < cnt; ++q) { // each group on its own
+		var id = sgmapback[q];
+		var sgroup = molecule.sgroups.get(id);
+		let sgDetails = `${this.getMol3000Prefix()}${id + 1} ${sgroup.type} 0`;
+		if (sgroup.atoms.length) {
+			// atoms that define the Sgroup
+			// ATOMS=(natoms aidx1 aidx2 ..)
+			sgDetails += ` ATOMS=(${sgroup.atoms.length} ${sgroup.atoms.map(atom => atom + 1).join(' ')})`;
```

Comment:
I assume that mol files are 1-indexed and the ketcher internal data structure is zero indexed.

Thought/Idea: It *might* be worth having some named util functions to do this? `toCtabIdx(ketcherIdx)`, and the inverse. I realize that the computation is incredibly trivial but there are *a lot* of `someVariable + 1` operations in this block of code... 

With named util functions you could express, e.g. `sgroup.atoms.map(toCtabAtomIdx)`.

Reply:
+1!

@pradeepnschrodinger  and I [discussed this offline](https://github.com/schrodinger/ketcher/pull/13#discussion_r386988111) to have a generic ticket to create utils for such purposes, and possibly perform some clean-up/refactor.

Reply:
ticket here https://jira.schrodinger.com/browse/SS-29875